From 8032e967a4684e93ab25067b1b4344dbd8d0526d Mon Sep 17 00:00:00 2001
From: Peter Korsgaard <jacmet@sunsite.dk>
Date: Sun, 31 May 2009 09:37:49 +0200
Subject: [PATCH 1/3] dbus-daemon-launch-helper: fix build with libxml

dbus-daemon-launch-helper links with the trival config parser, which
doesn't provide bus_config_parser_check_doctype(), like the big parser does.
The libxml loader unfortunately wants to call it, so this breaks the build.

Fix it by including a dummy bus_config_parser_check_doctype() in the trivial
config parser.

Signed-off-by: Peter Korsgaard <jacmet@sunsite.dk>
Signed-off-by: Marc Kleine-Budde <mkl@pengutronix.de>
---
# 20091211 rsc: mailed to dbus mailing list
# 20100825 wsa: better version mainline: http://cgit.freedesktop.org/dbus/dbus/commit/?id=085e7730b88a0c4a42a325c3ed4dd99f0b67185b

 bus/config-parser-trivial.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/bus/config-parser-trivial.c b/bus/config-parser-trivial.c
index 98d554e..a936371 100644
--- a/bus/config-parser-trivial.c
+++ b/bus/config-parser-trivial.c
@@ -310,6 +310,14 @@ bus_config_parser_get_service_dirs (BusConfigParser *parser)
   return &parser->service_dirs;
 }
 
+dbus_bool_t
+bus_config_parser_check_doctype (BusConfigParser   *parser,
+                                 const char        *doctype,
+                                 DBusError         *error)
+{
+    return TRUE;
+}
+
 #ifdef DBUS_BUILD_TESTS
 #include <stdio.h>
 #include "test.h"
-- 
1.6.3.1

